fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2490,6 +2493,9 @@ public LoanOverdueDTO applyChargeToOverdueLoanInstallment(final Long loanId, fin
 
                 final LoanCharge loanCharge = LoanCharge.createNewFromJson(loan, chargeDefinition, command, entry.getValue());
 
+                if (BigDecimal.ZERO.compareTo(loanCharge.amount()) == 0) {
+                    return null;
```

Comment:
Currently we have only these types of `ChargeCalculationType`.
```    FLAT(1, "chargeCalculationType.flat"), //
    PERCENT_OF_AMOUNT(2, "chargeCalculationType.percent.of.amount"), //
    PERCENT_OF_AMOUNT_AND_INTEREST(3, "chargeCalculationType.percent.of.amount.and.interest"), //
    PERCENT_OF_INTEREST(4, "chargeCalculationType.percent.of.interest"),
    PERCENT_OF_DISBURSEMENT_AMOUNT(5,"chargeCalculationType.percent.of.disbursement.amount");
```
Even though it seems, that the amount can be different for a different iteration, in fact it is always the same.

Reply:
Ok that may be right I'm just saying it as I don't take anything for granted and also it will avoid returning a null value as the overdueDTO.

Also, something similiar it could happen in the applyOverdueChargesForLoan function with the if statement:

if(overdueDTO == null){
     return;
}

Because you are iterating a list of multiple overdue installments for the same loan. If the penalty charge is calculated based on the installment interest amount, that amount can be different for each one of those installments, for the first one the calculated amount for all the installment charges can be 0, but for the next ones it could be a value greater than zero. Making a return call would not calculate charges for the next overdue installments.

Reply:
I have made those changes.

Reply:
It seems right. Thanks